src/plots/gl2d/scene2d.js

```diff
@@ -268,6 +268,16 @@ proto.updateFx = function(options) {
     fullLayout.hovermode = options.hovermode;
 };
 
+var relayoutCallback = function(scene) {
+    var update = {};
+    update[scene.id] = { // scene.camera has no many useful projection or scale information
+        lastInputTime: scene.camera.lastInputTime, // helps determine which one is the latest input (if async)
+        xrange: scene.xaxis.range.slice(0),
+        yrange: scene.yaxis.range.slice(0)
+    };
+    scene.container.parentElement.parentElement.parentElement.emit('plotly_relayout', update);
```

Comment:
we should pass a ref to the `gd` when create to 2d scene [here](https://github.com/plotly/plotly.js/blob/d8e27dd33270aa7e4bb01a612cc99c51fa3cb68a/src/plots/gl2d/index.js#L53) as we do for 3d scenes [there](https://github.com/plotly/plotly.js/blob/d8e27dd33270aa7e4bb01a612cc99c51fa3cb68a/src/plots/gl3d/index.js#L55).